### PR TITLE
modify to support registering CR by CID and DID

### PR DESCRIPTION
**Pull request by RainFallsSilent**

### CODE REVIEW

File: common/config/params.go [hunk 1]
```diff
@@ -393,6 +396,10 @@ type Params struct {
 	// statistics error.
 	VoteStatisticsHeight uint32
 
+	// RegisterCRByDIDHeight defines the height to support register and update
+	// CR by CID and DID.
+	RegisterCRByDIDHeight uint32
```
Comment (peterwillcn) [inline common/config/params.go]: Will the name be changed to RegisterCRByCIDHeight?

Comment (RainFallsSilent) [reply common/config/params.go]: Starting from this height would support registering CR by did, so I think RegisterCRByDIDHeight is better.

File: settings.go [hunk 2]
```diff
@@ -665,6 +665,12 @@ func newSettings() *settings {
 		ConfigPath:   "CRConfiguration.VotingPeriod",
 		ParamName:    "CRVotingPeriod"})
 
+	result.Add(&settingItem{
+		Flag:         cmdcom.RegisterCRByDIDHeightFlag,
+		DefaultValue: uint32(0),
+		ConfigPath:   "CRConfiguration.RegisterCRByDIDHeightFlag",
```
Comment (Houshoupei84) [inline settings.go]: RegisterCRByDIDHeightFlag -> RegisterCRByDIDHeight

Comment (RainFallsSilent) [reply settings.go]: fixed

**Pull request merged**